src/IRC/Types.hs

```diff
@@ -37,3 +39,19 @@ data IrcParams = IrcParams { host     :: Network
 data Hook = Hook { onConnect    :: Connection -> IO ()
                  , onEvent      :: [IrcEvent]
                  , onDisconnect :: Connection -> IO () }
+
+data Options = Options { network            :: Network
```

Comment:
I think the `Options` type should stay in the `Main` module as it defines the interface of it (via CLI) and could in the future hold options that are not at all related to IRC.